### ISSUE TITLE
"F841 local variable unused" triggering erroniously

### ISSUE THREAD

**Issue by jhallard** [charliermarsh/ruff#889]
Example code below:

```
    # Verify the annotation is coming from two different sources
    annotations = get_annotations(tdm_client, node_uid)

    assert len({annotations["source_uid"] for annotations in annotations}) == 2
```

This is triggering for me

```
.../.../test_annotation_filter.py:352:5: F841 Local variable `annotations` is assigned to but never used
```

**Comment by charliermarsh** [comment]
Thanks for this! Will fix today.

**Comment by jhallard** [comment]
Thank you!!

**Comment by charliermarsh** [comment]
Going out in [v0.0.136](https://github.com/charliermarsh/ruff/releases/tag/v0.0.136) -- sorry for the delay :)

**Comment by jhallard** [comment]
Incredible turn around.. we just adopted `ruff` internally at my company and are seeing 50x speedups over out flake8, black, and isort lint stack!

**Comment by charliermarsh** [comment]
@jhallard - Amazing! Don't hesitate to file issues or feature requests as they come up, would love to hear them and definitely aim to be very attentive to early adopters :)

(Separately, I've never used the enterprise product but was always a big fan of the ideas around Snorkel! I built a lot of infra and tooling for data labeling + active learning loops when I was at [Spring](https://www.springdiscovery.com/).)
